doc/userdoc/contribute/styleguide/styleguide.rst

```diff
@@ -0,0 +1,654 @@
+   Output is directly below input without any brackets.
+   A blank line must end the example.
+
+For in-text code use the role :code: or double back ticks::
+
+   ``cout << "Hello, World!`` or
+   :code:`cout << "Hello, World!"`
+
+.. _math_style:
+
+Math equations
+~~~~~~~~~~~~~~
+
+The input language for mathematics is LaTeX markup. See `Mathematics into Type
+<http://www.ams.org/arc/styleguide/mit-2.pdf>`_ for a guide to styling LaTeX math.
+
+
+For equations that take a whole line (or more), use the math directive::
+
+    .. math::
+
+        f(x) = \int_{-\infty}^{\infty} \hat{f}(\xi) e^{2 \pi i x \xi} \, d\xi.
+
+Output rendered as
+
+    .. math::
+
+        f(x) = \int_{-\infty}^{\infty} \hat{f}(\xi) e^{2 \pi i x \xi} \, d\xi.
+
+If the equation runs over several lines you can use double backslashes ``\\`` as a separator at the end of each line.
+You can also align lines in an equation by using the ``&`` where you want an equation aligned::
+
+
+    .. math::
+
+         (a + b)^2  &=  (a + b)(a + b) \\
+                    &=  a^2 + 2ab + b^2
+
+Rendered as
+
+    .. math::
+
+         (a + b)^2  &=  (a + b)(a + b) \\
+                    &=  a^2 + 2ab + b^2
+
+For in-text math, use the math role::
+
+   Now we can see :math:`x=1` for this example.
+
+Rendered as
+
+   Now we can see :math:`x=1` for this example.
+
+.. _sec_admonition:
+
+Admonitions
+~~~~~~~~~~~
+
+Admonitions are directives that render as highlighted blocks to draw the reader's attention to a particular point.
+
+Use them sparingly.
+
+
+Use the admonition
+
+* "See also" to reference internal or external links (only in cases where the reference should stand out),
+
+* "Note" to add additional information that the reader needs to be aware of,
+
+* "Warning" to indicate that something might go wrong without the provided information, and
+
+* "Danger" if the situation may cause severe, possibly irreversible, problems.
+
+
+If you want a custom admonition, use:
+
+.. code-block:: none
+
+   .. admonition:: Custom label
+
+      Here is some text
+
+Rendered as
+
+
+   .. admonition:: Custom label
+
+         Here is some text
+
+
+References
+~~~~~~~~~~
+
+For referencing reStructuredText files within the documentation, use the ``:doc:`` role. It requires the relative path to
+the file::
+
+   :doc:`sample_doc`
+
+In this case, the link text will be the title of the given document:
+
+   :doc:`sample_doc`
+
+You can specify the text you want to use for the link by doing the following::
+
+   :doc:`custom label <sample_doc>`
+
+This will be rendered as
+
+   :doc:`custom label <sample_doc>`
+
+For cross-referencing specific section headings, figures, or other arbitrary places within a file, use the ``:ref:`` role.
+
+The ``:ref:`` role requires a reference label that looks like this ``.. _type_ref-label:``.
+
+::
+
+   .. _sec_my-ref-label:
+
+   Section to cross-reference
+   --------------------------
+
+   Some content in this section.
+
```

Comment:
This seems to render the "in" as code. Should this be a `.. code-block:: text` or `.. code-block:: null` ?

Reply:
Fixed by my recent suggestion.